### ISSUE TITLE
gnome.generate_gir() not using pkg-config bin path

### ISSUE THREAD

**Issue by adrianjohnson** [mesonbuild/meson#2400]
I've installed a later version of gir in a custom directory. PKG_CONFIG_PATH is setup for this directory. meson finds the new version of gir but uses the system gir instead of the gir in the bindir of the gobject-introspection-1.0 pkg-config.

```
Program g-ir-scanner found: YES (/usr/bin/g-ir-scanner)
Program g-ir-compiler found: YES (/usr/bin/g-ir-compiler)
Native dependency gobject-introspection-1.0 found: YES 1.54.0
```
1.54 is the new version. System vesion is 1.5.0.

**Comment by nirbheek** [comment]
We should use the `g_ir_scanner` and `g_ir_compiler`, etc pkg-config variables inside `gobject-introspection-1.0`.

**Comment by ebassi** [comment]
Working on this, in light of a series of changes we're trying to include [inside GLib itself](https://gitlab.gnome.org/GNOME/glib/merge_requests/375).

See also the discussion [on GitLab](https://gitlab.gnome.org/GNOME/glib/issues/1521).

**Comment by grimmy** [comment]
The problem I'm seeing is that the `PKG_CONFIG_PATH` environment variable, either from the system (via `export`) or via the new(-ish) `--pkg-config-path` command line argument to meson is not used in `generate_gir`.
I've done a bit of digging so far, but haven't figured out why it's not in the environment in the state that's passed to `generate_gir` which calls `PkgConfigDependency` while it works fine for other `dependency` calls.

**Comment by eli-schwartz** [comment]
fixed piecemeal by #6687, amended in #6696, made to support subprojects in #7020 (at which time it finally began respecting pkg-config first), and fixed yet again in #7072 to respect cross file wrapper overrides
